specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-06-02-preview/fleets.json

```diff
@@ -0,0 +1,817 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create a fleet member resource. Joins an existing cluster to the fleet": {
+            "$ref": "./examples/FleetMembers_Create.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "FleetMembers"
+        ],
+        "operationId": "FleetMembers_Get",
+        "summary": "Gets a Fleet member.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetMemberNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FleetMember"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets a fleet member resource.": {
+            "$ref": "./examples/FleetMembers_Get.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "FleetMembers"
+        ],
+        "operationId": "FleetMembers_Delete",
+        "summary": "Deletes a fleet member.",
+        "description": "Deleting a Fleet member results in the member cluster leaving fleet. The Member azure resource is deleted upon success. The underlying cluster is not deleted.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetMemberNameParameter"
+          },
+          {
+            "$ref": "#/parameters/IfMatchHeader"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

Need to add 200 for the case the resource existed prior to the delete request.

Reply:
asynchronous. same as other deletes, we use 202 according to the RPC guideline

Reply:
202: async response
204: resource was previously deleted/does not exist
200: resource was deleted by the request
Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource)

Reply:
But there is no cases where our api returns 200. You cannot delete this resource synchronously, we are always cleaning up something async, so it's always 202.

Reply:
Per the async contract the Url returned in the location header will return 200/204 when the async request is complete. Ref: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously. The protocol is that the client will poll the Url returned in the location header until something other than 202 is returned. It's true that is a different endpoint, but correct SDK generation depends on including those status codes here. It's kind of a hack, but that's how it can work without requiring all RPs to document their async status endpoints. You can see many/most other specs do the same.

Reply:
we use the OperationResult pattern, so the header we return is Azure-AsyncOperation, with a URL to the async operation api.
Not a url to this resource.

Reply:
ah, I see, the Location is marked as MUST, while the operation is marked as Optional.
We followed conventions that AKS implemented, so we did not add the Location header here.
but, how are we supposed to differentiate the 204 case from the 200 case.
when the resource is gone, we don't have any state anymore that tells us that this operation was the one that initiated the delete.
polling the Azure-AsyncOperation url (the operations api) is much simpler, since the operation record is still there after the resource was deleted.

Reply:
adding 200